### PR TITLE
Improved local protobuf generation

### PR DESCRIPTION
**Pull request by izaaklauer**
When I attempt `make gen`, I get the following error:

```
protoc-gen-go: invalid Go import path "." for "plugin.proto"

The import path must contain at least one forward slash ('/') character.

See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.

--go_out: protoc-gen-go: Plugin failed with status code 1.
```

And though that message isn't especially clear, the solution (from [here](https://github.com/techschool/pcbook-go/issues/3#issuecomment-823206034)) is this change.

Please take a stab at `make gen` on this branch and let me know if this breaks anything for you.

### CODE REVIEW

**Review by krantzinator — APPROVED** [review]
`make gen` works, though I get a large diff -- I think because we haven't upgraded the protoc-gen-go version in this repo yet to 1.26.0?
```
$ git diff --stat
 proto/gen/plugin.pb.go | 857 +++++++++++++++++++++++++++++++++++++++---------------------------------------
 proto/gen/unused.pb.go |  13 +-
 2 files changed, 430 insertions(+), 440 deletions(-)
```

**Comment by krantzinator** [comment]
I probably was too quick to approve this, re: slack discussion. For the sake of documentation here, I think if we merge this in, we should do so with the updated gens as well. Waypoint is on protoc-gen-go 1.26.0, and protoc 1.5.2.

**Comment by izaaklauer** [comment]
It seems the problem with updating the go_package line alone is that it will break generating protobufs with the old packages defined here.

To solve this, I've upgraded the protobuf packages to the same versions specified in core waypoint. I've also copied the `tools` pattern from core, so now running `make tools` will install the versions of the protobuf chain that are declared in go.mod.

The next time someone has a protobuf version problem, hopefully running `make tools` will sort them out.

**Comment by mitchellh** [comment]
Before we merge this let me hop in and try to make sure the `shell.nix` stuff is right so that that continues to work since I use that to get my tools versions.

**Review by krantzinator — APPROVED** [review]
Beautiful

File: nix/sources.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "owner": "nmattia",
+        "repo": "niv",
+        "rev": "527494090f7075ed5e3aa15ef7093846e5e25d52",
+        "sha256": "1hahf0i0iix1iyzaaq5a7rbgyfm3xjcfl69lm1b5mas1p8vdimih",
+        "type": "tarball",
+        "url": "https://github.com/nmattia/niv/archive/527494090f7075ed5e3aa15ef7093846e5e25d52.tar.gz",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
+    },
+    "nixpkgs": {
+        "branch": "nixpkgs-unstable",
```
Comment (izaaklauer) [inline nix/sources.json:15]: Stole this from core waypoint - the default stable branch of nixpkgs doesn't seem to have go 1.16.

**Comment by izaaklauer** [comment]
> Before we merge this let me hop in and try to make sure the `shell.nix` stuff is right so that that continues to work since I use that to get my tools versions.

Oh whoops, I mis-read this and took a stab at updating the nix stuff myself. I have no stake in the nix setup, so feel free to overwrite my changes if they're not what you had in mind.

**Review by briancain — APPROVED** [review]
I can't test any of the nix stuff but the rest looks good 😄

**Comment by mitchellh** [comment]
> > Before we merge this let me hop in and try to make sure the `shell.nix` stuff is right so that that continues to work since I use that to get my tools versions.
> 
> Oh whoops, I mis-read this and took a stab at updating the nix stuff myself. I have no stake in the nix setup, so feel free to overwrite my changes if they're not what you had in mind.

Oh I didn't see you did that. I'll give it a shot today!

**Pull request merged**